reference/promise-types/reports.markdown

```diff
@@ -112,24 +112,20 @@ Include part of a file in a report.
 
 **Default value:** `5`
 
-**Allowed input range:** `0,99999999999`
+**Allowed input range:** `-99999999999,99999999999`
```

Comment:
hurmmmmmm.
Feels like this should be a macro.

Reply:
We have function prototype macro, and we have a promise attribute macro, perhaps we lack a body attribute macro.

Reply:
scanning the code it looks like promise_attribute knows about body attributes.

```
	if header[2].find("Attributes") != -1:
		# assumption for promise type definition
		# header[1] = promise type
		# header[2] = "Attributes"
		# header[3] = attribute name
		# header[4] = body attribute
		promise_type_def = promise_types[header[1]]
		attribute_def = promise_type_def["attributes"][header[3]]
		attribute_type = attribute_def["type"]
		if attribute_type == "body":
			if header[4]:
				body_type_def = body_types[header[3]]
				attribute_def = body_type_def["attributes"][header[4]]
				attribute_type = attribute_def["type"]
			else:
				lines.append("**Type:** `body %s`\n\n" % header[3])
				return lines

```

Sooooooo, I think we should try 

```
[%CFEngine_promise_attribute(5)%]
```
And that hopefully then renders Type, default value and allowed input range.

Reply:
It didnt :(